From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 12 Dec 2018 01:16:28 +0000
Subject: ARM: Avoid ABI change in 4.9.139
Forwarded: not-needed

Commit 901e325f772f "ARM: bugs: add support for per-processor bug
checking" added a new member to struct processor.  This structure
is always instantiated in built-in code and the new member is only
used in built-in code.  Therefore we can safely add the new member
at the end instead.

Move it to the end and hide it from genksyms.  Also hide it when
building modules, to make sure they really don't use it.

---

Index: linux/arch/arm/include/asm/proc-fns.h
===================================================================
--- linux.orig/arch/arm/include/asm/proc-fns.h
+++ linux/arch/arm/include/asm/proc-fns.h
@@ -37,10 +37,6 @@ extern struct processor {
 	 */
 	void (*_proc_init)(void);
 	/*
-	 * Check for processor bugs
-	 */
-	void (*check_bugs)(void);
-	/*
 	 * Disable any processor specifics
 	 */
 	void (*_proc_fin)(void);
@@ -79,6 +75,12 @@ extern struct processor {
 	unsigned int suspend_size;
 	void (*do_suspend)(void *);
 	void (*do_resume)(void *);
+#if !defined(__GENKSYMS__) && !defined(MODULE)
+	/*
+	 * Check for processor bugs
+	 */
+	void (*check_bugs)(void);
+#endif
 } processor;
 
 #ifndef MULTI_CPU
Index: linux/arch/arm/mm/proc-macros.S
===================================================================
--- linux.orig/arch/arm/mm/proc-macros.S
+++ linux/arch/arm/mm/proc-macros.S
@@ -281,7 +281,6 @@ ENTRY(\name\()_processor_functions)
 	.word	\dabort
 	.word	\pabort
 	.word	cpu_\name\()_proc_init
-	.word	\bugs
 	.word	cpu_\name\()_proc_fin
 	.word	cpu_\name\()_reset
 	.word	cpu_\name\()_do_idle
@@ -309,6 +308,8 @@ ENTRY(\name\()_processor_functions)
 	.word	0
 	.endif
 
+	.word	\bugs
+
 	.size	\name\()_processor_functions, . - \name\()_processor_functions
 .endm
 
